### ISSUE TITLE
Checks configuration fot the `stylecheck` linter is ignored

### ISSUE THREAD

**Issue by antoineco** [golangci/golangci-lint#2281]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

Setting the value of stylecheck's `checks` parameter to `["all"]` (or any variation with single quotes / without quotes) does not have the desired effect. The output returns no error, while errors were expected.

For comparison, running the linter directly with the same configuration on the included snippet returns the expected warnings:

```console
$ cat staticcheck.conf
checks = ["all"]
```

```console
$ staticcheck ./foo/foo.go
foo/foo.go:1:1: at least one file in a package should have a package comment (ST1000)
foo/foo.go:3:1: comment on exported type MyType should be of the form "MyType ..." (with optional leading article) (ST1021)
foo/foo.go:6:1: comment on exported function DoStuff should be of the form "DoStuff ..." (ST1020)
```

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.42.1 built from 54f4301d on 2021-09-06T17:05:10Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
linters:
  enable:
  - stylecheck

linters-settings:
  stylecheck:
    checks:
    - all
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.17.1 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/acotten/.cache/go-build"
GOENV="/home/acotten/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/acotten/go/pkg/mod"
GOOS="linux"
GOPATH="/home/acotten/go"
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.17.1"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/home/acotten/go/src/github.com/triggermesh/triggermesh/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build2232512087=/tmp/go-build -gno-record-gcc-switches"
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
(no output)
```

```console
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /home/acotten/go/src/github.com/triggermesh/triggermesh/foo /home/acotten/go/src/github.com/triggermesh/triggermesh /home/acotten/go/src/github.com/triggermesh /home/acotten/go/src/github.com /home/acotten/go/src /home/acotten/go /home/acotten /home /]
INFO [config_reader] Used config file .golangci.yaml
INFO [lintersdb] Active 12 linters: [deadcode errcheck gofmt gosimple govet ineffassign staticcheck structcheck stylecheck typecheck unused varcheck]
INFO [loader] Go packages loading at mode 575 (files|name|exports_file|imports|types_sizes|compiled_files|deps) took 26.1751ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 219µs
INFO [linters context/goanalysis] analyzers took 6.5116ms with top 10 stages: fact_deprecated: 730.1µs, buildir: 497.4µs, directives: 431.2µs, ST1000: 311.6
µs, deadcode: 296.3µs, tokenfileanalyzer: 258.9µs, ctrlflow: 247.9µs, printf: 204.9µs, asmdecl: 203.4µs, gofmt: 162.5µs
INFO [runner] Issues before processing: 3, after processing: 0
INFO [runner] Processors filtering stat (out/in): skip_dirs: 3/3, autogenerated_exclude: 3/3, exclude: 3/3, exclude-rules: 0/3, cgo: 3/3, filename_unadjuster: 3/3, identifier_marker: 3/3, path_prettifier: 3/3, skip_files: 3/3
INFO [runner] processing took 243.7µs with stages: identifier_marker: 102.9µs, exclude-rules: 80µs, path_prettifier: 24.4µs, autogenerated_exclude: 22.8µs, skip_dirs: 6.3µs, cgo: 1.4µs, uniq_by_line: 1µs, nolint: 1µs, filename_unadjuster: 900ns, skip_files: 600ns, max_same_issues: 500ns, exclude: 400ns, max_per_file_from_linter: 300ns, diff: 300ns, source_code: 300ns, path_shortener: 200ns, max_from_linter: 200ns, severity-rules: 100ns, path_prefixer: 100ns, sort_results: 0s
INFO [runner] linters took 47.1864ms with stages: goanalysis_metalinter: 46.8232ms
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 2 samples, avg is 26.2MB, max is 28.6MB
INFO Execution took 84.5882ms
```

</details>

### Code example or link to a public repository

<details>

```go
package foo

// Wrong doc format.
type MyType struct{}

// Wrong doc format.
func DoStuff(t *MyType) {}
```

</details>

**Comment by antoineco** [comment]
I stand corrected. Hitting `ST1000` will shadow **all** other `ST*` due to the default `EXC0011` exclusion.

I can't see how that's not a bug because I should be able to _not_ set a package comment, and still see legitimate `stylecheck` errors.
